cmlreaders/path_finder.py

```diff
@@ -151,9 +154,15 @@ def _get_most_recent_ramulator_folder(self, base_folder_path):
         timestamped_directories = glob.glob(os.path.join(self.rootdir,
                                                          base_folder_path))
 
-        # Sort such that most recently modified appears first
-        timestamped_directories = sorted(timestamped_directories,
-                                         key=os.path.getmtime, reverse=True)
+        # Remove all invalid names (valid = only contains numbers and _)
+        timestamped_directories = [
+            d for d in timestamped_directories
+            if os.path.isdir(d) and
+            all([c in string.digits for c in os.path.basename(d).replace('_', '')])
+        ]
+
+        # Sort such that most recent appears first
```

Comment:
I'm assuming this does alpha-numeric sorting so the most recent folder should pop to the surface?

Reply:
It removes all entries that don't contain only digits and underscore since that's how the Ramulator directories are named. Then the only sorting is by number and `sorted` doesn't need any additional info to DTRT.